src/bentoml/_internal/resource.py

```diff
@@ -183,120 +188,95 @@ def query_os_cpu_count() -> int:
     return 1
 
 
-# class MemResource(Resource, resource_id="mem"):
-#     @classmethod
-#     def from_spec(cls, spec: t.Any):
-#         assert isinstance(mem, (int, str)), "mem must be int or str"
-#
-#         if isinstance(mem, int):
-#             return mem
-#
-#         unit_match = re.match("([0-9]+)([A-Za-z]{1,2})", mem)
-#         mem_multipliers = {
-#             "k": 1000,
-#             "M": 1000**2,
-#             "G": 1000**3,
-#             "T": 1000**4,
-#             "P": 1000**5,
-#             "E": 1000**6,
-#             "Ki": 1024,
-#             "Mi": 1024**2,
-#             "Gi": 1024**3,
-#             "Ti": 1024**4,
-#             "Pi": 1024**5,
-#             "Ei": 1024**6,
-#         }
-#         if unit_match:
-#             base = int(unit_match[1])
-#             unit = unit_match[2]
-#             if unit in mem_multipliers:
-#                 return base * mem_multipliers[unit]
-#         raise ValueError(f"Invalid MEM resource limit '{mem}'")
-
-
-class NvidiaGpuResource(Resource[t.List[int]], resource_id="nvidia.com/gpu"):
+class NvidiaGpuResource(Resource[t.List[str]], resource_id="nvidia.com/gpu"):
     @classmethod
-    def from_spec(cls, spec: int | str | list[int | str]) -> list[int]:
+    def from_spec(cls, spec: int | str | list[int | str]) -> list[str]:
         if not isinstance(spec, (int, str, list)):
             raise TypeError(
                 "NVidia GPU device IDs must be int, str or a list specifing the exact GPUs to use."
             )
 
         try:
             if isinstance(spec, int):
+                if spec == -1:
+                    return []
                 if spec < -1:
                     raise ValueError
-                return list(range(spec))
+                return [str(i) for i in range(spec)]
             elif isinstance(spec, str):
-                return cls.from_spec(int(spec))
+                try:
+                    resolved = int(spec)
+                    return cls.from_spec(resolved)
+                except ValueError:
+                    if spec.startswith("GPU"):
```

Comment:
I believe we should also support `MIG` strings. Let's maybe limit parsing to as little as possible so we don't have to keep up with changes in the Nvidia libraries.